ghci-wrapper/test/Language/Haskell/GhciWrapperSpec.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE CPP #-}
+module Language.Haskell.GhciWrapperSpec (main, spec) where
+
+import           Test.Hspec
+import           System.IO.Silently
+
+import           Control.Exception
+import           Data.List (isSuffixOf)
+
+import           Language.Haskell.GhciWrapper (Interpreter, Config(..), defaultConfig)
+import qualified Language.Haskell.GhciWrapper as Interpreter
+
+main :: IO ()
+main = hspec spec
+
+withInterpreterConfig :: Config -> (Interpreter -> IO a) -> IO a
+withInterpreterConfig config action = bracket (Interpreter.new config []) Interpreter.close $ \ghci -> action ghci
+
+withInterpreter :: ((String -> IO String) -> IO a) -> IO a
+withInterpreter action = withInterpreterConfig defaultConfig $ \ghci -> action (Interpreter.eval ghci)
+
+shouldEvaluateTo :: (Show a, Eq a) => IO a -> a -> IO ()
+action `shouldEvaluateTo` expected = action >>= (`shouldBe` expected)
+
+spec :: Spec
+spec = do
+  describe "evalEcho" $ do
+    it "prints result to stdout" $ do
+      withInterpreterConfig defaultConfig $ \ghci -> do
+        (capture $ Interpreter.evalEcho ghci ("putStr" ++ show "foo\nbar")) `shouldReturn` ("foo\nbar", "foo\nbar")
+
+  describe "eval" $ do
+    it "shows literals" $ withInterpreter $ \ghci -> do
+      ghci "23" `shouldEvaluateTo` "23\n"
```

Comment:
Technically, there is a difference between `putStr "foo"` and `putStrLn "foo"`.  I think it won't hurt to preserve this distinction in `ghci-wrapper`.  Doctest removes it by doing a `lines` on the output, so both `"foo"` and `"foo\n"` will become `["foo"]`.